File: SystemLibCE/SystemLibSampleCECS/Backup/Form_LED.cs
//=============================================================================
// CASIO Mobile Device SystemLib Sample                                       +
// Copyright (C) 2006 CASIO COMPUTER CO., LTD. All Rights Reserved.           +
//=============================================================================
// Form_LED.cs : LED sample                                                        +
//-----------------------------------------------------------------------------
using System;
using System.Drawing;
using System.Collections;
using System.ComponentModel;
using System.Windows.Forms;
using Calib;


namespace SystemLibSampleCS
{
	/// <summary>
	/// This is a form of the LED sample.
	/// </summary>
	public class Form_LED : System.Windows.Forms.Form
	{
		private System.Windows.Forms.Label label1;
		private System.Windows.Forms.Label label2;
		private System.Windows.Forms.Label label3;
		private System.Windows.Forms.TextBox textBox_Num;
		private System.Windows.Forms.TextBox textBox_OnTime;
		private System.Windows.Forms.TextBox textBox_OffTime;
		private System.Windows.Forms.Button button_Green;
		private System.Windows.Forms.Button button_Red;
		private System.Windows.Forms.Button button_Orange;
		private System.Windows.Forms.Button button_Magenta;
		private System.Windows.Forms.Button button_Cyan;
		private System.Windows.Forms.Button button_Blue;
		private System.Windows.Forms.Button button_Off;
		private System.Windows.Forms.Button button_Get;
	
		public Form_LED()
		{
			//
			// Required for Windows Form Designer support
			//
			InitializeComponent();


			//
			// TODO: Add any constructor code after InitializeComponent call
			//
		}

		/// <summary>
		/// Clean up any resources being used.
		/// </summary>
		protected override void Dispose( bool disposing )
		{
			base.Dispose( disposing );
		}

		#region Windows Form Designer generated code
		/// <summary>
		/// Required method for Designer support - do not modify
		/// the contents of this method with the code editor.
		/// </summary>
		private void InitializeComponent()
		{
			this.label1 = new System.Windows.Forms.Label();
			this.label2 = new System.Windows.Forms.Label();
			this.label3 = new System.Windows.Forms.Label();
			this.textBox_Num = new System.Windows.Forms.TextBox();
			this.textBox_OnTime = new System.Windows.Forms.TextBox();
			this.textBox_OffTime = new System.Windows.Forms.TextBox();
			this.button_Green = new System.Windows.Forms.Button();
			this.button_Red = new System.Windows.Forms.Button();
			this.button_Orange = new System.Windows.Forms.Button();
			this.button_Magenta = new System.Windows.Forms.Button();
			this.button_Cyan = new System.Windows.Forms.Button();
			this.button_Blue = new System.Windows.Forms.Button();
			this.button_Off = new System.Windows.Forms.Button();
			this.button_Get = new System.Windows.Forms.Button();
			// 
			// label1
			// 
			this.label1.Location = new System.Drawing.Point(16, 16);
			this.label1.Size = new System.Drawing.Size(88, 16);
			this.label1.Text = "LED ON count";
			// 
			// label2
			// 
			this.label2.Location = new System.Drawing.Point(16, 48);
			this.label2.Size = new System.Drawing.Size(88, 16);
			this.label2.Text = "LED ON time";
			// 
			// label3
			// 
			this.label3.Location = new System.Drawing.Point(16, 80);
			this.label3.Size = new System.Drawing.Size(88, 16);
			this.label3.Text = "LED OFF Time";
			// 
			// textBox_Num
			// 
			this.textBox_Num.Location = new System.Drawing.Point(136, 8);
			this.textBox_Num.Size = new System.Drawing.Size(88, 19);
			this.textBox_Num.Text = "1";
			// 
			// textBox_OnTime
			// 
			this.textBox_OnTime.Location = new System.Drawing.Point(136, 40);
			this.textBox_OnTime.Size = new System.Drawing.Size(88, 19);
			this.textBox_OnTime.Text = "16";
			// 
			// textBox_OffTime
			// 
			this.textBox_OffTime.Location = new System.Drawing.Point(136, 72);
			this.textBox_OffTime.Size = new System.Drawing.Size(88, 19);
			this.textBox_OffTime.Text = "16";
			// 
			// button_Green
			// 
			this.button_Green.Location = new System.Drawing.Point(16, 104);
			this.button_Green.Size = new System.Drawing.Size(64, 24);
			this.button_Green.Text = "Green";
			this.button_Green.Click += new System.EventHandler(this.button_Green_Click);
			// 
			// button_Red
			// 
			this.button_Red.Location = new System.Drawing.Point(88, 104);
			this.button_Red.Size = new System.Drawing.Size(64, 24);
			this.button_Red.Text = "Red";
			this.button_Red.Click += new System.EventHandler(this.button_Red_Click);
			// 
			// button_Orange
			// 
			this.button_Orange.Location = new System.Drawing.Point(160, 104);
			this.button_Orange.Size = new System.Drawing.Size(64, 24);
			this.button_Orange.Text = "Orange";
			this.button_Orange.Click += new System.EventHandler(this.button_Orange_Click);
			// 
			// button_Magenta
			// 
			this.button_Magenta.Location = new System.Drawing.Point(160, 136);
			this.button_Magenta.Size = new System.Drawing.Size(64, 24);
			this.button_Magenta.Text = "Magenta";
			this.button_Magenta.Click += new System.EventHandler(this.button_Magenta_Click);
			// 
			// button_Cyan
			// 
			this.button_Cyan.Location = new System.Drawing.Point(88, 136);
			this.button_Cyan.Size = new System.Drawing.Size(64, 24);
			this.button_Cyan.Text = "Cyan";
			this.button_Cyan.Click += new System.EventHandler(this.button_Cyan_Click);
			// 
			// button_Blue
			// 
			this.button_Blue.Location = new System.Drawing.Point(16, 136);
			this.button_Blue.Size = new System.Drawing.Size(64, 24);
			this.button_Blue.Text = "Blue";
			this.button_Blue.Click += new System.EventHandler(this.button_Blue_Click);
			// 
			// button_Off
			// 
			this.button_Off.Location = new System.Drawing.Point(16, 176);
			this.button_Off.Size = new System.Drawing.Size(208, 24);
			this.button_Off.Text = "LED OFF";
			this.button_Off.Click += new System.EventHandler(this.button_Off_Click);
			// 
			// button_Get
			// 
			this.button_Get.Location = new System.Drawing.Point(16, 208);
			this.button_Get.Size = new System.Drawing.Size(208, 24);
			this.button_Get.Text = "Get status";
			this.button_Get.Click += new System.EventHandler(this.button_Get_Click);
			this.FormBorderStyle = System.Windows.Forms.FormBorderStyle.FixedToolWindow;
			// 
			// Form_LED
			// 
			this.ClientSize = new System.Drawing.Size(240, 246);
			this.Controls.Add(this.button_Get);
			this.Controls.Add(this.button_Off);
			this.Controls.Add(this.button_Magenta);
			this.Controls.Add(this.button_Cyan);
			this.Controls.Add(this.button_Blue);
			this.Controls.Add(this.button_Orange);
			this.Controls.Add(this.button_Red);
			this.Controls.Add(this.button_Green);
			this.Controls.Add(this.textBox_OffTime);
			this.Controls.Add(this.textBox_OnTime);
			this.Controls.Add(this.textBox_Num);
			this.Controls.Add(this.label3);
			this.Controls.Add(this.label2);
			this.Controls.Add(this.label1);
			this.MaximizeBox = false;
			this.MinimizeBox = false;
			this.Text = "LED";
			this.Closed += new System.EventHandler(this.button_Off_Click);

		}
		#endregion

		private void button_Green_Click(object sender, System.EventArgs e)
		{
			SetLED(Calib.SystemLibNet.Def.LED_GREEN);
		}

		private void button_Red_Click(object sender, System.EventArgs e)
		{
			SetLED(Calib.SystemLibNet.Def.LED_RED);		
		}

		private void button_Orange_Click(object sender, System.EventArgs e)
		{
			SetLED(Calib.SystemLibNet.Def.LED_ORANGE);		
		}

		private void button_Blue_Click(object sender, System.EventArgs e)
		{
			SetLED(0x0000000B);
		}

		private void button_Cyan_Click(object sender, System.EventArgs e)
		{
			SetLED(0x0000000C);		
		}

		private void button_Magenta_Click(object sender, System.EventArgs e)
		{
			SetLED(0x0000000D);
		}

		private void button_Off_Click(object sender, System.EventArgs e)
		{
			SetLED(Calib.SystemLibNet.Def.LED_OFF);
		}

		private void button_Get_Click(object sender, System.EventArgs e)
		{
			int	dwRet;

			// get LED ON/OFF condition
			dwRet = Calib.SystemLibNet.Api.SysGetLED();
			switch(dwRet) 
			{
				case Calib.SystemLibNet.Def.LED_GREEN:
					MessageBox.Show("Green","SysGetLED",MessageBoxButtons.OK,MessageBoxIcon.None,MessageBoxDefaultButton.Button1);
					break;
				case Calib.SystemLibNet.Def.LED_RED:
					MessageBox.Show("Red","SysGetLED",MessageBoxButtons.OK,MessageBoxIcon.None,MessageBoxDefaultButton.Button1);
					break;
				case Calib.SystemLibNet.Def.LED_ORANGE:
					MessageBox.Show("Orange","SysGetLED",MessageBoxButtons.OK,MessageBoxIcon.None,MessageBoxDefaultButton.Button1);
					break;
				case 0x0000000B:
					MessageBox.Show("Blue","SysGetLED",MessageBoxButtons.OK,MessageBoxIcon.None,MessageBoxDefaultButton.Button1);
					break;
				case 0x0000000C:
					MessageBox.Show("Cyan","SysGetLED",MessageBoxButtons.OK,MessageBoxIcon.None,MessageBoxDefaultButton.Button1);
					break;
				case 0x0000000D:
					MessageBox.Show("Magenta","SysGetLED",MessageBoxButtons.OK,MessageBoxIcon.None,MessageBoxDefaultButton.Button1);
					break;
			}
		}

		private void SetLED(int dwMode)
		{
			int	dwNum, dwOnTime, dwOffTime;

			// LED ON count get
			dwNum = Convert.ToInt32(textBox_Num.Text);

			// LED ON time get
			dwOnTime = Convert.ToInt32(textBox_OnTime.Text);

			// LED OFF time get
			dwOffTime = Convert.ToInt32(textBox_OffTime.Text);

			// LED ON/OFF
			Calib.SystemLibNet.Api.SysSetLED(dwMode,dwNum, dwOnTime, dwOffTime);
		}

	}
}

File: SystemLibCE/SystemLibSampleCEVB/Form_LED.vb
'=============================================================================
' CASIO Mobile Device SystemLib Sample                                       +
' Copyright (C) 2006 CASIO COMPUTER CO., LTD. All Rights Reserved.           +
'=============================================================================
' Form_LED.vb : LED sample                                                        +
'-----------------------------------------------------------------------------
Public Class Form_LED
    Inherits System.Windows.Forms.Form
    Friend WithEvents textBox_Num As System.Windows.Forms.TextBox
    Friend WithEvents label3 As System.Windows.Forms.Label
    Friend WithEvents label2 As System.Windows.Forms.Label
    Friend WithEvents label1 As System.Windows.Forms.Label
    Friend WithEvents textBox_OffTime As System.Windows.Forms.TextBox
    Friend WithEvents textBox_OnTime As System.Windows.Forms.TextBox
    Friend WithEvents button_Get As System.Windows.Forms.Button
    Friend WithEvents button_Off As System.Windows.Forms.Button
    Friend WithEvents button_Magenta As System.Windows.Forms.Button
    Friend WithEvents button_Cyan As System.Windows.Forms.Button
    Friend WithEvents button_Blue As System.Windows.Forms.Button
    Friend WithEvents button_Orange As System.Windows.Forms.Button
    Friend WithEvents button_Red As System.Windows.Forms.Button
    Friend WithEvents button_Green As System.Windows.Forms.Button


#Region " Windows Form Designer generated code "

    Public Sub New()
        MyBase.New()

        'This call is required by the Windows Form Designer.
        InitializeComponent()

        'Add any initialization after the InitializeComponent() call

    End Sub

    'Form overrides dispose to clean up the component list.
    Protected Overloads Overrides Sub Dispose(ByVal disposing As Boolean)
        MyBase.Dispose(disposing)
    End Sub

    'NOTE: The following procedure is required by the Windows Form Designer
    'It can be modified using the Windows Form Designer.  
    'Do not modify it using the code editor.
    <System.Diagnostics.DebuggerStepThrough()> Private Sub InitializeComponent()
        Me.textBox_Num = New System.Windows.Forms.TextBox
        Me.label3 = New System.Windows.Forms.Label
        Me.label2 = New System.Windows.Forms.Label
        Me.label1 = New System.Windows.Forms.Label
        Me.textBox_OffTime = New System.Windows.Forms.TextBox
        Me.textBox_OnTime = New System.Windows.Forms.TextBox
        Me.button_Get = New System.Windows.Forms.Button
        Me.button_Off = New System.Windows.Forms.Button
        Me.button_Magenta = New System.Windows.Forms.Button
        Me.button_Cyan = New System.Windows.Forms.Button
        Me.button_Blue = New System.Windows.Forms.Button
        Me.button_Orange = New System.Windows.Forms.Button
        Me.button_Red = New System.Windows.Forms.Button
        Me.button_Green = New System.Windows.Forms.Button
        '
        'textBox_Num
        '
        Me.textBox_Num.Location = New System.Drawing.Point(144, 8)
        Me.textBox_Num.Size = New System.Drawing.Size(88, 19)
        Me.textBox_Num.Text = "1"
        '
        'label3
        '
        Me.label3.Location = New System.Drawing.Point(8, 64)
        Me.label3.Size = New System.Drawing.Size(88, 16)
        Me.label3.Text = "LED OFF time"
        '
        'label2
        '
        Me.label2.Location = New System.Drawing.Point(8, 40)
        Me.label2.Size = New System.Drawing.Size(88, 16)
        Me.label2.Text = "LED ON time"
        '
        'label1
        '
        Me.label1.Location = New System.Drawing.Point(8, 16)
        Me.label1.Size = New System.Drawing.Size(88, 16)
        Me.label1.Text = "LED ON count"
        '
        'textBox_OffTime
        '
        Me.textBox_OffTime.Location = New System.Drawing.Point(144, 56)
        Me.textBox_OffTime.Size = New System.Drawing.Size(88, 19)
        Me.textBox_OffTime.Text = "16"
        '
        'textBox_OnTime
        '
        Me.textBox_OnTime.Location = New System.Drawing.Point(144, 32)
        Me.textBox_OnTime.Size = New System.Drawing.Size(88, 19)
        Me.textBox_OnTime.Text = "16"
        '
        'button_Get
        '
        Me.button_Get.Location = New System.Drawing.Point(16, 200)
        Me.button_Get.Size = New System.Drawing.Size(208, 24)
        Me.button_Get.Text = "Get status"
        '
        'button_Off
        '
        Me.button_Off.Location = New System.Drawing.Point(16, 168)
        Me.button_Off.Size = New System.Drawing.Size(208, 24)
        Me.button_Off.Text = "LED OFF"
        '
        'button_Magenta
        '
        Me.button_Magenta.Location = New System.Drawing.Point(160, 128)
        Me.button_Magenta.Size = New System.Drawing.Size(64, 24)
        Me.button_Magenta.Text = "Magenta"
        '
        'button_Cyan
        '
        Me.button_Cyan.Location = New System.Drawing.Point(88, 128)
        Me.button_Cyan.Size = New System.Drawing.Size(64, 24)
        Me.button_Cyan.Text = "Cyan"
        '
        'button_Blue
        '
        Me.button_Blue.Location = New System.Drawing.Point(16, 128)
        Me.button_Blue.Size = New System.Drawing.Size(64, 24)
        Me.button_Blue.Text = "Blue"
        '
        'button_Orange
        '
        Me.button_Orange.Location = New System.Drawing.Point(160, 96)
        Me.button_Orange.Size = New System.Drawing.Size(64, 24)
        Me.button_Orange.Text = "Orange"
        '
        'button_Red
        '
        Me.button_Red.Location = New System.Drawing.Point(88, 96)
        Me.button_Red.Size = New System.Drawing.Size(64, 24)
        Me.button_Red.Text = "Red"
        '
        'button_Green
        '
        Me.button_Green.Location = New System.Drawing.Point(16, 96)
        Me.button_Green.Size = New System.Drawing.Size(64, 24)
        Me.button_Green.Text = "Green"
        '
        'Form_LED
        '
        Me.ClientSize = New System.Drawing.Size(240, 238)
        Me.Controls.Add(Me.button_Get)
        Me.Controls.Add(Me.button_Off)
        Me.Controls.Add(Me.button_Magenta)
        Me.Controls.Add(Me.button_Cyan)
        Me.Controls.Add(Me.button_Blue)
        Me.Controls.Add(Me.button_Orange)
        Me.Controls.Add(Me.button_Red)
        Me.Controls.Add(Me.button_Green)
        Me.Controls.Add(Me.textBox_OffTime)
        Me.Controls.Add(Me.textBox_OnTime)
        Me.Controls.Add(Me.label3)
        Me.Controls.Add(Me.label2)
        Me.Controls.Add(Me.label1)
        Me.Controls.Add(Me.textBox_Num)
        Me.MaximizeBox = False
        Me.MinimizeBox = False
        Me.Text = "Form_LED"

    End Sub

#End Region

    Private Sub button_Green_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles button_Green.Click
        SetLED(Calib.SystemLibNet.Def.LED_GREEN)
    End Sub

    Private Sub button_Red_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles button_Red.Click
        SetLED(Calib.SystemLibNet.Def.LED_RED)
    End Sub

    Private Sub button_Orange_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles button_Orange.Click
        SetLED(Calib.SystemLibNet.Def.LED_ORANGE)
    End Sub

    Private Sub button_Blue_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles button_Blue.Click
        SetLED(&HB)
    End Sub

    Private Sub button_Cyan_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles button_Cyan.Click
        SetLED(&HC)
    End Sub

    Private Sub button_Magenta_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles button_Magenta.Click
        SetLED(&HD)
    End Sub

    Private Sub button_Off_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles button_Off.Click
        SetLED(Calib.SystemLibNet.Def.LED_OFF)
    End Sub

    Private Sub button_Get_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles button_Get.Click
        Dim dwRet As Int32

        ' get LED ON/OFF condition
        dwRet = Calib.SystemLibNet.Api.SysGetLED()
        Select Case dwRet
            Case Calib.SystemLibNet.Def.LED_GREEN
                MessageBox.Show("Green", "SysGetLED", MessageBoxButtons.OK, MessageBoxIcon.None, MessageBoxDefaultButton.Button1)
            Case Calib.SystemLibNet.Def.LED_RED
                MessageBox.Show("Red", "SysGetLED", MessageBoxButtons.OK, MessageBoxIcon.None, MessageBoxDefaultButton.Button1)
            Case Calib.SystemLibNet.Def.LED_ORANGE
                MessageBox.Show("Orange", "SysGetLED", MessageBoxButtons.OK, MessageBoxIcon.None, MessageBoxDefaultButton.Button1)
            Case &HB
                MessageBox.Show("Blue", "SysGetLED", MessageBoxButtons.OK, MessageBoxIcon.None, MessageBoxDefaultButton.Button1)
            Case &HC
                MessageBox.Show("Cyan", "SysGetLED", MessageBoxButtons.OK, MessageBoxIcon.None, MessageBoxDefaultButton.Button1)
            Case &HD
                MessageBox.Show("Magenta", "SysGetLED", MessageBoxButtons.OK, MessageBoxIcon.None, MessageBoxDefaultButton.Button1)
        End Select

    End Sub

    Private Sub SetLED(ByVal dwMode As Int32)
        Dim dwNum As Int32, dwOnTime As Int32, dwOffTime As Int32

        ' LED ON count get
        dwNum = Convert.ToInt32(textBox_Num.Text)

        ' LED ON time get
        dwOnTime = Convert.ToInt32(textBox_OnTime.Text)

        ' LED OFF time get
        dwOffTime = Convert.ToInt32(textBox_OffTime.Text)

        ' LED ON/OFF
        Calib.SystemLibNet.Api.SysSetLED(dwMode, dwNum, dwOnTime, dwOffTime)

    End Sub

    Protected Overridable Sub Form_LED_OnClosed(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles MyBase.Closed

		' LED OFF then close dialog
        Calib.SystemLibNet.Api.SysSetLED(Calib.SystemLibNet.Def.LED_OFF, 0, 0, 0)
        button_Off_Click(sender, e)
    End Sub
End Class
